yamoveset/panels/main.py

```diff
@@ -368,3 +375,69 @@ def on_paste(self, _):
         # with MultiMessageDialog(
         #         self, 'The following entries in the listed files have been changed:', msg, changed_msg, wx.OK) as dlg:
         #     dlg.ShowModal()
+
+    def on_add(self, _):
+
+
+        if not self.parent.copied:
+            return
+
+        copied = pickle.loads(self.parent.copied)
+
+
+
+        selected_data = [item for item in copied]
+
+
+
+        # Paste entries
+        selected_values = []
+        copied_values = []
+        changed_values = defaultdict(list)
+
+        for n, copied_data in enumerate(copied):
+            selected_values.append((selected_data[n].index, selected_data[n].get_static_values()))
+            copied_values.append(copied_data.get_static_values())
+
+        for selected_val, copied_val in zip(selected_values, copied_values):
+            # Example:
+            # Item type: Animation
+            # entry: Index
+            # dependency: Type
+            # depend_value: 5
+            # entry_values: {1, 2, 3}
+            for item_type, v1 in copied_val.items():
+                if item_type not in [Animation, Hitbox, Camera]:
+                    continue
+                for entry_pair, v2 in v1.items():
+                    for depend_value, entry_values in v2.items():
+                        # Skip if dependency isn't a character
+                        if item_type.dependencies[entry_pair][depend_value] != 'Character':
+                            continue
+                        if not self.get_changed_values(
+                                changed_values, item_type, entry_pair, depend_value, entry_values,
+                                selected_val, selected_data):
+                            return

```

Comment:
This entire block originally was more for ensuring that values related to ean/bdm didn't get messed up when pasting over entries.  Since this is adding, comparing to the selected entries doesn't really matter.

Reply:
Also with this change, get ready for an influx of movesets that are bloated with duplicate EANs.

Reply:
i removed this block of code originally but that resulted in adding the entry with no changes (as in, no new animations or BDM entries added)
with this here, it basically does the same thing as pasting entries
and if an animation/BDM entry already exists, it doesn't add it again so no duplicates. at least from my testing

Reply:
ok, yeah, I remember nothing of what I did back then on what trickery I did do the check.  I only remember self.links which it uses to make sure duplicate EAN/BDM entries don't get added within the same session.